kontent-ai/sample-app-next-js
# DEVREL-977 - Flatten circular data, fix product filtering

### Motivation

[DEVREL-977](https://kontent-ai.atlassian.net/jira/software/projects/DEVREL/boards/421?selectedIssue=DEVREL-977)

- replacement PR for https://github.com/kontent-ai/sample-app-next-js/pull/56

Adds helper methods to flatten/restore items with potential circular references to pass through getStaticProps. Fixes a bug where product filtering wasn't applied at all.

### Checklist


### How to test

Introduce a circular reference in Kontent.ai project (e.g. via content item link in any given Fact item), run the app and visit any page that requests said item, confirm it works as expected.
